x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -190,6 +198,27 @@ public SemanticTextFieldMapper build(MapperBuilderContext context) {
                 builderParams(this, context)
             );
         }
+
+        /**
+         * As necessary, copy settings from this builder to the passed-in mapper.
+         * Used to preserve {@link SemanticTextField.ModelSettings} when updating a semantic text mapping.
+         *
+         * @param mapper The mapper
+         * @return A mapper with the copied settings applied
+         */
+        private SemanticTextFieldMapper copySettings(SemanticTextFieldMapper mapper, MapperMergeContext mapperMergeContext) {
+            SemanticTextFieldMapper returnedMapper = mapper;
+            if (mapper.fieldType().getModelSettings() == null && mapper.fieldType().getEmbeddingsField() == null) {
```

Comment:
Do we need to check the embeddings field too? Is there a path where model settings is null but not the embeddings field?

Reply:
Probably not. This is an extra defensive check for mappers emitted from the inference on ingest code path. In that code path, both the model settings and embeddings field should be set. However, if we're concerned with only one of those being set, it's better to check closer to the source of the potential problem. 

I will simplify this check and investigate if we should add an assertion/error check somewhere else for this case.